### PR TITLE
DOC: fix issue with __new__ usage in subclassing doc.

### LINKED ISSUE #14386: bug in documentation example code for subclassing ndarray
There seems to be a minor bug in the example code for subclassing ndarray [in this section of the online documentation](https://docs.scipy.org/doc/numpy/user/basics.subclassing.html#a-brief-python-primer-on-new-and-init)

The example for class C includes a call to `object.__new__()` with more than one argument:

```python
class C(object):
    def __new__(cls, *args):
        print('Cls in __new__:', cls)
        print('Args in __new__:', args)
        return object.__new__(cls, *args)
```

Calling this results in an error:

```
  File "/home/jon/python/ndarray_testing.py", line 31, in __new__
    return object.__new__(cls, *args)

TypeError: object.__new__() takes exactly one argument (the type to instantiate)
```

If the extra argument to `object.__new__()` is removed, the example works as expected.

### Fixed example:

```python
class C(object):
    def __new__(cls, *args):
        print('Cls in __new__:', cls)
        print('Args in __new__:', args)
        return object.__new__(cls)

    def __init__(self, *args):
        print('type(self) in __init__:', type(self))
        print('Args in __init__:', args)
```

### Output of fixed example
```
In [129]: C('foo')
Cls in __new__: <class '__main__.C'>
Args in __new__: ('foo',)
type(self) in __init__: <class '__main__.C'>
Args in __init__: ('foo',)
Out[129]: <__main__.C at 0x7f27fb972710>
```

### Numpy/Python version information:

1.16.2 3.7.3 (default, Mar 27 2019, 22:11:17) 
[GCC 7.3.0]

### PR DESCRIPTION
**Pull request by rgommers**
Closes gh-14386.

### CODE REVIEW

**Comment by charris** [comment]
Maybe worth a comment? There may be potential confusion between generic python objects and the Python `object` type. I'm not even clear on how to express the difference. Something like
```
# The object type __new__ method takes a single argument.
```

**Comment by Aa88s** [comment]
# The object type __new__ method takes a single argument.
fixed it.

File: numpy/doc/subclassing.py [hunk 1]
```diff
@@ -118,7 +118,8 @@ class C(object):
       def __new__(cls, *args):
           print('Cls in __new__:', cls)
           print('Args in __new__:', args)
-          return object.__new__(cls, *args)
+          # The `object` type __new__ method takes a single argument.
```
[the remark this answers is not in the document]
Comment (mattip) [reply numpy/doc/subclassing.py]: ping

[the remark this answers is not in the document]
Comment (eric-wieser) [reply numpy/doc/subclassing.py]: Or maybe just `# do not forward any arguments to object.__new__`

**Comment by charris** [comment]
Rewording doesn't make enough difference to keep this open. Thanks Ralf.

**Pull request merged**